### PR TITLE
Fix: Add SMTrainingCompilerConfigurationError to the list of registered exception classes.

### PR DESCRIPTION
**Pull request by vinayburugu**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-training-toolkit/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-training-toolkit/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have used the regional endpoint when creating S3 and/or STS clients (if appropriate)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-training-toolkit/blob/master/README.md)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker_training/process.py [hunk 1]
```diff
@@ -82,6 +82,15 @@ def get_tensorflow_exception_classes():
     return exception_classes
 
 
+def get_pytorch_exception_classes():
+    """Pytorch configuration errors are raised by pytorch_xla.py."""
+    exception_classes = []
+    exception_classes += [errors.SMTrainingCompilerConfigurationError.__class__.__name__]
```
Comment (Lokiiiiii) [inline src/sagemaker_training/process.py:88]: String conversion is automatic, does not have to be explicit

File: src/sagemaker_training/process.py [hunk 2]
```diff
@@ -82,10 +82,10 @@ def get_tensorflow_exception_classes():
     return exception_classes
 
 
-def get_pytorch_exception_classes():
+def get_trainingcompiler_exception_classes():
     """Pytorch configuration errors are raised by pytorch_xla.py."""
     exception_classes = []
-    exception_classes += [errors.SMTrainingCompilerConfigurationError.__class__.__name__]
+    exception_classes += [errors.SMTrainingCompilerConfigurationError.__class__]
```
Comment (Lokiiiiii) [inline src/sagemaker_training/process.py:88]:
```suggestion
    exception_classes += [errors.SMTrainingCompilerConfigurationError]
```

**Review by Lokiiiiii — CHANGES_REQUESTED** [review]

File: src/sagemaker_training/process.py [hunk 3]
```diff
@@ -82,6 +82,15 @@ def get_tensorflow_exception_classes():
     return exception_classes
 
 
+def get_trainingcompiler_exception_classes():
+    """Pytorch configuration errors are raised by pytorch_xla.py."""
+    exception_classes = []
+    exception_classes += [errors.SMTrainingCompilerConfigurationError__name__]
```
Comment (Lokiiiiii) [inline src/sagemaker_training/process.py:88]:
```suggestion
    exception_classes += [errors.SMTrainingCompilerConfigurationError]
```

**Review by Lokiiiiii — CHANGES_REQUESTED** [review]

File: src/sagemaker_training/process.py [hunk 4]
```diff
@@ -85,7 +85,7 @@ def get_tensorflow_exception_classes():
 def get_trainingcompiler_exception_classes():
     """Pytorch configuration errors are raised by pytorch_xla.py."""
     exception_classes = []
-    exception_classes += [errors.SMTrainingCompilerConfigurationError__name__]
+    exception_classes += [errors.SMTrainingCompilerConfigurationError.__name__]
```
Comment (Lokiiiiii) [inline src/sagemaker_training/process.py:88]: Remove __name__ . It is not required.

**Review by Lokiiiiii — DISMISSED** [review]

File: src/sagemaker_training/process.py (hunk 4 above)
Comment (Lokiiiiii) [reply src/sagemaker_training/process.py:88]: Please refer to how debugger has done it

File: src/sagemaker_training/process.py [hunk 5]
```diff
@@ -82,6 +82,15 @@ def get_tensorflow_exception_classes():
     return exception_classes
 
 
+def get_trainingcompiler_exception_classes():
+    """Pytorch configuration errors are raised by pytorch_xla.py."""
+    exception_classes = []
+    exception_classes += [errors.SMTrainingCompilerConfigurationError]
+    if not exception_classes:
```
Comment (josephevans) [inline src/sagemaker_training/process.py:89]: Won't this always be true considering the lines above?

**Review by josephevans — APPROVED** [review]

**Pull request merged**